debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/buffered/MemoryTransactionCacheAdapter.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.logminer.buffered;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import io.debezium.connector.oracle.logminer.buffered.memory.MemoryLogMinerTransactionCache;
+import io.debezium.connector.oracle.logminer.buffered.memory.MemoryTransaction;
+import io.debezium.connector.oracle.logminer.events.LogMinerEvent;
+
+/**
+ * An adapter that maps provider transactions to internal MemoryTransaction instances.
+ * Single-threaded since LogMiner processing is single-threaded.
+ */
+public class MemoryTransactionCacheAdapter<T extends Transaction> {
+
+    private final MemoryLogMinerTransactionCache inner;
+    private final Map<String, MemoryTransaction> txMap = new HashMap<>();
```

Comment:
```suggestion
    private final Map<String, MemoryTransaction> transactionsById = new HashMap<>();
```
Just to keep with the naming in the other caches.